core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -178,6 +178,23 @@ class HistoryServer(
     provider.getListing()
   }
 
+  /**
+    * Returns a list of available applications, in descending order according to their end time.
+    *
+    * @param limit the number of applications to return
+    * @return List of known applications with a limit.
+    */
+  def getApplicationInfoList(limit: Int): Iterator[ApplicationInfo] = {
```

Comment:
If you use Integer.MaxValue to mean unlimited, then just the call to limit() works as intended.